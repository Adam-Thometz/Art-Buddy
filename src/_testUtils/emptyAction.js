const EMPTY_ACTION = { type: "" }

export default EMPTY_ACTION