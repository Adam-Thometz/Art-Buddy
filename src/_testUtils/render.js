import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
