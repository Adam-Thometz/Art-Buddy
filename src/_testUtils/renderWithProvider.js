import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;