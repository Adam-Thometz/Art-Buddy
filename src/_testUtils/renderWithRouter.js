import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;