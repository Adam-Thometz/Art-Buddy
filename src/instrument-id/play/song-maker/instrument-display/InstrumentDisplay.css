.InstrumentDisplay {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(2, 1fr);
  gap: 50px;
  width: 67%;
  height: 80%;
}

.InstrumentDisplay-options {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
}

.InstrumentDisplay-remove {
  cursor: pointer;
  text-decoration: underline;
}