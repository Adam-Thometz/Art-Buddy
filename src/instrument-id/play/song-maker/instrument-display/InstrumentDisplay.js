import { useSelector, useDispatch } from "react-redux";
import {
  addInstrument,
  removeInstrument,
  selectInstrument,
  selectMelody,
} from "_redux/instrument-id/song-maker/songMakerActions";

import "./InstrumentDisplay.css";

import AddIcon from "_components/icon/add-icon/AddIcon";
import Dropdown from "_components/dropdown/Dropdown";

import {
  instrumentOptions,
  melodyOptions,
  rhythmOptions,
} from "./dropdownOptions";

const InstrumentDisplay = () => {
  const { song } = useSelector((state) => state.songMaker);
  const dispatch = useDispatch();

  const handleAddInstrument = (e) => {
    const id = +e.currentTarget.id;
    dispatch(addInstrument(id));
  };

  const handleRemoveInstrument = (e) => {
    const id = +e.currentTarget.id;
    dispatch(removeInstrument(id));
  };

  const handleSelectInstrument = (e) => {
    const id = +e.currentTarget.id;
    const instrumentId = e.target.id;
    dispatch(selectInstrument({ id, instrumentId }));
  };

  const handleSelectMelody = (e) => {
    const id = +e.currentTarget.id;
    const melodyId = e.target.id;
    dispatch(selectMelody({ id, melodyId }));
  };

  return (
    <div className="InstrumentDisplay">
      {song.map((instrument, i) => {
        if (!instrument) {
          return (
            <AddIcon
              key={i}
              text="ADD INSTRUMENT"
              size="69px"
              id={i}
              onClick={handleAddInstrument}
            />
          );
        } else {
          const instrumentDropdown = (
            <Dropdown
              key={`instrument${i}`}
              id={i}
              labelText="INSTRUMENT"
              options={instrumentOptions}
              onClick={handleSelectInstrument}
            />
          );
          const melodyDropdown = !instrument.instrumentId ? null : (
            <Dropdown
              key={`melody${i}`}
              id={i}
              labelText={instrument.isRhythm ? "RHYTHM" : "MELODY"}
              options={instrument.isRhythm ? rhythmOptions : melodyOptions}
              onClick={handleSelectMelody}
            />
          );
          return (
            <div key={i} className="InstrumentDisplay-options">
              {instrumentDropdown}
              {melodyDropdown}
              <span
                className="InstrumentDisplay-remove"
                id={i}
                onClick={handleRemoveInstrument}
              >
                remove
              </span>
            </div>
          );
        }
      })}
    </div>
  );
};

export default InstrumentDisplay;
