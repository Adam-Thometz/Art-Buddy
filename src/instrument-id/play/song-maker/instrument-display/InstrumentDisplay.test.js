import InstrumentDisplay from "./InstrumentDisplay";

import { render } from "_testUtils/render";

jest.mock('tone', () => ({
  Buffer: jest.fn()
}));

describe('InstrumentDisplay component', () => {
  it('renders without crashing', () => {
    render(<InstrumentDisplay />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<InstrumentDisplay />);
    expect(asFragment()).toMatchSnapshot();
  });
});