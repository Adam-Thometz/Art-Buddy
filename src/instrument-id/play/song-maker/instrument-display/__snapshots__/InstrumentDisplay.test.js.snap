// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstrumentDisplay component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="InstrumentDisplay"
  >
    <div
      class="InstrumentDisplay-options"
    >
      <div
        class="Dropdown"
      >
        <span
          class="Dropdown-label"
          style="z-index: 5;"
        >
          INSTRUMENT 
          <img
            alt=""
            src="file"
          />
        </span>
      </div>
      <span
        class="InstrumentDisplay-remove"
        id="0"
      >
        remove
      </span>
    </div>
    <figure
      class="Icon"
      data-testid=""
      id="1"
    >
      <img
        alt=""
        src="file"
        style="height: 69px; width: 69px; cursor: pointer;"
      />
      <figcaption
        class="Icon-text"
        style="font-size: 1.75rem;"
      >
        ADD INSTRUMENT
      </figcaption>
    </figure>
    <figure
      class="Icon"
      data-testid=""
      id="2"
    >
      <img
        alt=""
        src="file"
        style="height: 69px; width: 69px; cursor: pointer;"
      />
      <figcaption
        class="Icon-text"
        style="font-size: 1.75rem;"
      >
        ADD INSTRUMENT
      </figcaption>
    </figure>
    <figure
      class="Icon"
      data-testid=""
      id="3"
    >
      <img
        alt=""
        src="file"
        style="height: 69px; width: 69px; cursor: pointer;"
      />
      <figcaption
        class="Icon-text"
        style="font-size: 1.75rem;"
      >
        ADD INSTRUMENT
      </figcaption>
    </figure>
  </div>
</DocumentFragment>
`;
